### PR TITLE
[FIX] tool-update: Compare tool versions correctly

### PR DESCRIPTION
**Pull request by edvilme**
Addresses #43089 and https://devdiv.visualstudio.com/DevDiv/_workitems/edit/2288480

When attempting to update a package to a new version, a check is made to ensure the same version is not installed twice. However, this check currently ignores the version suffix, which means users cannot update between versions with different release tags.

### CODE REVIEW

**Review by dsplaisted — APPROVED** [review]
Good fix!  Is it possible to add a test covering this case?

File: test/dotnet.Tests/CommandTests/ToolUpdateGlobalOrToolPathCommandTests.cs [hunk 1]
```diff
@@ -200,6 +200,20 @@ public void GivenAnExistedLowerversionInstallationWhenUpdateAllItCanPrintSuccess
                 _packageId, LowerPackageVersion, HigherPackageVersion));
         }
 
+        [Fact]
+        public void GivenAnExistedPreviewVersionInstallationWhenUpdateToHigherVersionItSucceeds()
+        {
+            CreateInstallCommand($"-g {_packageId} --version {HigherPreviewPackageVersion} --verbosity minimal");
+            _reporter.Lines.Clear();
```
Comment (dsplaisted) [inline test/dotnet.Tests/CommandTests/ToolUpdateGlobalOrToolPathCommandTests.cs:207]: Did you run the test to make sure it failed without your code change?  It looks like you may be skipping executing the install command here.

Comment (edvilme) [reply test/dotnet.Tests/CommandTests/ToolUpdateGlobalOrToolPathCommandTests.cs]: Oh youre right, I totally missed that

**Review by dsplaisted — APPROVED** [review]

**Comment by v-sherryfan** [comment]
@dsplaisted Will this fix be backported to .NET 9.0? It still blocks NET9.0 updrage tooling scenario.

**Comment by dsplaisted** [comment]
@marcpopMSFT Can we consider this for 9.0 servicing?  If not, we should at least put it in 9.0.200 I think.

**Comment by baronfel** [comment]
Would +1 backporting this.

**Comment by marcpopMSFT** [comment]
Yeah, seems like it's broken for every prerelease version. I don't know how common that is for tools but seems like a pretty critical scenario. Do we know when it broke though? edvilme indicated it's been this way for a while. If a while, then 9.0.2xx. If recent, 9.0.1xx.

**Comment by marcpopMSFT** [comment]
That makes it a regression in 9 so I'd support porting to 9.0.101.

**Pull request merged**